Assets/Scripts/EntityController.cs

```diff
@@ -58,12 +67,62 @@ protected void Movement(Vector2 target, Vector2 offset)
         rb.velocity = Vector2.ClampMagnitude(rb.velocity, maxSpeed);
     }
 
+    protected void launchArrow(EntityController target)
```

Comment:
Sorry, but arrows needs some major changes.
Using `transform.find`, the arrow logic being tied to the entity instead of being independent, the sprite always existing... there's some issues here that, if you were to try and add more projectiles in the future, it just wouldn't be easily feasible.

So I'll try to break this down to the best of my abilities, but feel free to ask for clarification anywhere:
- As a projectile, an arrow is independent of its source the moment its launched. In other words, best practice here would be to create an arrow prefab. This prefab contains whatever components are required (e.g. sprite renderer, probably a trigger collision as well) and a script with its logic.
- The arrow's logic does not have to be complicated. Something as simple as moving forward, and despawning when either its lifetime ends or when it hits something is enough. Properties like homing, like you have here, can be interesting, but at least that shouldn't be a part of a simple arrow.
- Spawning and resetting a projectile can be `Instantiate()` and `Destroy()`, respectively. In the long run this would need to be substituted by an Object Pooling system for performance, but you can ignore these concerns for now.
    - The instancing would be done by the source of the arrow, while the destruction would be by the arrow itself
    - You can have the instancing be rather generic if you leverage the `UnitData` objects, but more about that in another comment